### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by apfelbox**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Docs PR       | —

* Fix source map support
* Change implementation to TypeScript
* Refactor code
* Change parameter names

### CODE REVIEW

File: src/Compiler.ts [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+     */
+    private async postProcess (css: CompilationResult, entry: CompilationEntry) : Promise<PostProcessingResult>
+    {
+        try
+        {
+            return await this.postProcessor.process(css.css, {
+                from: entry.src,
+                to: entry.outFilePath,
+                map: {
+                    annotation: false, //this.options.debug,
```
Comment (keichinger) [inline src/Compiler.ts]: Do we need the commented value or is it old?

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+     * Callback on when a file has changed
+     */
+    private onChangedFile (file: string) : void
+    {
+        if (this.options.lint)
+        {
+            this.compiler.lint(file);
+        }
+
+        this.compileAll(false);
```
Comment (keichinger) [inline src/index.ts:142]: Can't we just call `this.compileAll(this.options.lint)`?

Comment (apfelbox) [reply src/index.ts:142]: No, as we only want to lint the changed file, not all files every time.

File: src/Compiler.ts (hunk 1 above)
Comment (apfelbox) [reply src/Compiler.ts]: Yes 😉

**Review by keichinger — APPROVED** [review]

**Pull request merged**